packages/components/src/date-time/style.scss

```diff
@@ -110,6 +110,12 @@
 			background: $light-gray-300;
 			border-color: $dark-gray-100;
 			box-shadow: inset 0 2px 5px -3px $dark-gray-500;
+			&:focus {
+				box-shadow:
+					inset 0 2px 5px -3px $dark-gray-500,
```

Comment:
Where do these values come from? `.components-datetime__time-am-button` is of type `Button`, so why do we not simply allow those focus styles to take effect? Or at least to be consistent.

https://github.com/WordPress/gutenberg/blob/6171097d4688fa28e739260616c157110f24e93f/packages/components/src/button/style.scss#L37-L47

Reply:
The _inset_ box-shadow value is the one used for the "pressed" state. It needs to be repeated for the `:focus` state otherwise the new box-shadow values will override it.

Reply:
> The _inset_ box-shadow value is the one used for the "pressed" state. It needs to be repeated for the `:focus` state otherwise the new box-shadow values will override it.

That makes sense, thanks. I suppose it may be a separate question then of whether there ought to be the notion of a "pressed" state as an option (prop?) for the `Button` component. I assume it would be styled roughly equivalent to [its `:active` styles](https://github.com/WordPress/gutenberg/blob/6171097d4688fa28e739260616c157110f24e93f/packages/components/src/button/style.scss#L49)? 

My main concern here is maintenance overhead and risk of inconsistencies via duplicated styles for what all ultimately describe a button's appearance.

Reply:
Looking at the `:active` style (on the left) and the "pressed" style:

<img width="117" alt="Screenshot 2019-05-13 at 21 09 46" src="https://user-images.githubusercontent.com/1682452/57647746-181b6880-75c4-11e9-9816-2c44037fa83f.png">

they're very similar and maybe the CSS part could be simplified.

The "pressed" state is used to represent the state of the underlying option via the `aria-pressed` attribute (based on the `isToggled` prop) and should not be used for states triggered by user interaction (hover/focus/active).